packaging/common/HalideHelpersConfig.cmake

```diff
@@ -2,10 +2,11 @@ cmake_minimum_required(VERSION 3.22)
 
 set(Halide_HOST_TARGET @Halide_HOST_TARGET@)
 
-include(CMakeFindDependencyMacro)
-
-set(THREADS_PREFER_PTHREAD_FLAG TRUE)
-find_dependency(Threads)
+if (NOT (DEFINED Halide_NO_THREADS AND Halide_NO_THREADS))
+    include(CMakeFindDependencyMacro)
+    set(THREADS_PREFER_PTHREAD_FLAG TRUE)
+    find_dependency(Threads)
+endif()
 
```

Comment:
If `Threads` is only needed for the runtime libraries and not for libHalide, then this logic should be moved to HalideGeneratorHelpers, in the if condition above.

It's a bit smelly here that the package config file references user-settable variables when determining which dependencies are needed by the package.